src/sonic-host-services-data/templates/common-password.j2

```diff
@@ -0,0 +1,43 @@
+#THIS IS AN AUTO-GENERATED FILE
+#
+# /etc/pam.d/common-password - password-related modules common to all services
+#
+# This file is included from other service-specific PAM config files,
+# and should contain a list of modules that define the services to be
+# used to change user passwords.  The default is pam_unix.
+
+# Explanation of pam_unix options:
+# The "yescrypt" option enables
+#hashed passwords using the yescrypt algorithm, introduced in Debian
+#11.  Without this option, the default is Unix crypt.  Prior releases
+#used the option "sha512"; if a shadow password hash will be shared
+#between Debian 11 and older releases replace "yescrypt" with "sha512"
+#for compatibility .  The "obscure" option replaces the old
+#`OBSCURE_CHECKS_ENAB' option in login.defs.  See the pam_unix manpage
+#for other options.
+
+# As of pam 1.0.1-6, this file is managed by pam-auth-update by default.
+# To take advantage of this, it is recommended that you configure any
+# local modules either before or after the default block, and use
+# pam-auth-update to manage selection of other modules.  See
+# pam-auth-update(8) for details.
+
+# here are the per-package modules (the "Primary" block)
+
+{% if passw_policies %}
+{% if passw_policies['state'] == 'enabled' %}
+password    requisite      pam_cracklib.so retry=3 maxrepeat=0 {% if passw_policies['len_min'] %}minlen={{passw_policies['len_min']}}{% endif %} {% if passw_policies['upper_class'] %}ucredit=-1{% else %}ucredit=0{% endif %} {% if passw_policies['lower_class'] %}lcredit=-1{% else %}lcredit=0{% endif %} {% if passw_policies['digits_class'] %}dcredit=-1{% else %}dcredit=0{% endif %} {% if passw_policies['special_class'] %}ocredit=-1{% else %}ocredit=0{% endif %} {% if passw_policies['reject_user_passw_match'] %}reject_username{% endif %} enforce_for_root
+
+password    required       pam_pwhistory.so {% if passw_policies['history_cnt'] %}remember={{passw_policies['history_cnt']}}{% endif %} use_authtok enforce_for_root
+{% endif %}
+{% endif %}
+
+password    [success=1 default=ignore]    pam_unix.so obscure yescrypt
```

Comment:
the logic is:
if the feature is enabled it will render the file including the pam_cracklib.so lines, 
if the feature state is disabled, It will not render the pam_cracklib.so lines, so in other words, the file will be rendered as Linux default (without passw policies ).

I used this flow because it is similar that what you coded in aaa files manipulation, you always render files from "scratch" according to the config_db configuration.

Reply:
pls see the comment above, I explained it in more detail.
In summary, the switch contains those 3 lines, so the idea is to not change the default values in the file.
Pls let me know if you agree, the file is located in /etc/pam.d/password-common.

Reply:
I see, here is current file content:


admin@vlab-01:/etc/pam.d$ grep "password" ./common-password
# /etc/pam.d/common-password - password-related modules common to all services
# used to change user passwords.  The default is pam_unix.
#hashed passwords using the yescrypt algorithm, introduced in Debian
#used the option "sha512"; if a shadow password hash will be shared
password        [success=1 default=ignore]      pam_unix.so obscure yescrypt
password        requisite                       pam_deny.so
password        required                        pam_permit.so
admin@vlab-01:/etc/pam.d$

Reply:
its a good match. Its exactly the same content like when the feature password-hardening is disabled.

Just when the feature state is enabled, it will append more information.